### PR TITLE
Add documentation on Hugo Drag and Drop

### PR DESCRIPTION
**Pull request by daut-morina**
This is my initial draft on the hugo drag and drop documentation. A couple of questions:

* ~What's the difference between agency and archive articles? There's an API documentation on hugo APIs. Unfortunately I could not find anything meaningful about those two types.~
* ~[This](https://github.com/nzzdev/livingdocs-api/blob/0afc4680eed6bf5c96131804ad154aa6460109b9/plugins/hugo-import-transformations/agency/util/transform.js#L5) - Does it have to be _resrc.it_?~
* ~The customers using this feature need the metadata plugin too, right?~

### CODE REVIEW

File: reference-docs/server-extensions/hugo-dnd.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Hugo Drag and Drop
+
+## Overview
+
+In case you use _Hugo_ as Digital Assets Management software, you can import articles from hugo by dragging and dropping them into livingdocs. There are two types of articles available for imports: _Agency_ and _Archive_ articles.
```
Comment (gabrielhase) [inline reference-docs/server-extensions/hugo-dnd.md]: The official name is `huGO+`

Comment (gabrielhase) [reply reference-docs/server-extensions/hugo-dnd.md]: An agency article is an article coming from aggregated agency feeds such as DPA, SDA, Reuters, etc.
Archive article are basically all the rest. Common cases are imports from the print system (print articles) or other third-party systems.

File: reference-docs/server-extensions/hugo-dnd.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+## Preparation
+
+Think about what types of articles you want to create from an imported hugo document, what design and layout you'd like them to have. You can specify any number of target documents with designs and layouts that are available to you.
+
+## Configuration
+
```
Comment (gabrielhase) [inline reference-docs/server-extensions/hugo-dnd.md]: please add some description what this configuration is used for. In any case the concept of the `basePath` should be explained and that you have scripts behind that

File: reference-docs/server-extensions/hugo-dnd.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+hugo:
+  targets:
+    basePath: path.resolve('./plugins/hugo-import-transformations')
+    # Hugo provides two types of articles, encoded as 'articleAgency' and 'articleArchive'
+    articleAgency:
+      dir: 'agency' # Specify the directory that contains transformations for agency articles
+      layouts: [ # Arbitrary number of targets possible
+        design: 'timeline' # You'd typically want to specify your own design here
+        layout: 'regular' # This can be any layout that is embedded in your design
+        transformation: 'regular' # This corresponds to a file named 'regular.js'
```
Comment (gabrielhase) [inline reference-docs/server-extensions/hugo-dnd.md]: needs more details what `regular.js` is supposed to be

File: reference-docs/server-extensions/hugo-dnd.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      ]
+```
+
+Every item in the configuration is required for the feature to work.
+
+## Transformations
+
+Now that you have configured the feature you'll want to provide transformations so the hugo document can be converted to a document that corresponds to your design and layout.
+
+A transformation is a single function that is expected to return an object containing a `livingdoc` and `metadata` and should have following signature:
```
Comment (gabrielhase) [inline reference-docs/server-extensions/hugo-dnd.md]: you probably want to link to the livingdoc documentation here: https://docs.livingdocs.io/reference-docs/common-livingdoc/livingdoc.html

File: reference-docs/server-extensions/hugo-dnd.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+You are provided with the `hugoArticle` you imported, the `design` and `layout` you have specified in your config and the `imagesApi` which is a service you can use to handle images.
+
+## Example transformation
+
+```js
+const async = require('async')
+const framework = require('@livingdocs/server/framework')
+
+module.exports = function ({hugoArticle, design, targetLayout, imagesApi}, callback) {
+  const imageService = conf.get('image_service') // You'll need to configure an imageservice like 'resrc.it' if you'd like to use images
```
Comment (gabrielhase) [inline reference-docs/server-extensions/hugo-dnd.md]: `resrc.it` is deprecated. Use `imgix` in the example

File: reference-docs/server-extensions/hugo-dnd.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+You are provided with the `hugoArticle` you imported, the `design` and `layout` you have specified in your config and the `imagesApi` which is a service you can use to handle images.
+
+## Example transformation
+
+```js
+const async = require('async')
+const framework = require('@livingdocs/server/framework')
+
+module.exports = function ({hugoArticle, design, targetLayout, imagesApi}, callback) {
```
Comment (gabrielhase) [inline reference-docs/server-extensions/hugo-dnd.md]: the example still uses the old API `targetLayout` -> `layout`

**Comment by peyerluk** [comment]
@daut-morina For the future please specify the release this change belongs to in the description.

I also crated a branch `releases/maintenance-2017-08` that the update can be merged into when reviewed if you finish the docs before the release is shipped.

**Comment by daut-morina** [comment]
Regarding the release: The changes had already been merged before we started working with maintenance branches. How do we proceed in this case?

**Comment by peyerluk** [comment]
Your changes landed in the release `maintenance-2017-08`, no?

**Comment by gabrielhase** [comment]
@daut-morina can you merge this one?

**Pull request merged**